OasisDEX/oasis-borrow
# Lw/7732 display followed aave vaults

# [AAVE Position Support for Follow Vaults on my positions page](https://app.shortcut.com/oazo-apps/story/7732/aave-position-support-for-follow-vaults-on-my-positions-page)

 
## Changes 👷‍♀️
- Followed AAVE positions are now displayed in followed table of each address
- Follow/Unfollow button is missing on Follow Table Rows because it requires refactoring all `DiscoveryTable` types and `Position`, to include protocol there and also possibly making supporting next protocols easier some of suggested changes are drafted on separate branch in this commit https://github.com/OasisDEX/oasis-borrow/pull/2147/commits/2aa2039f5370ab0cadc67dd88669f3487447a5f0
- Works with both AAVEV2 and AAVEV3 Positions
![image](https://user-images.githubusercontent.com/6871923/224083504-805f3383-6b4c-4b71-86f0-fc5fddac69ee.png)


## How to test 🧪
- Follow/Unfollow multiple Maker and Aave 2 and Aave V3 positions
- All Maker and Aave positions are correctly displayed in Followed Table with proper data
- Double check the data in FollowTable and in position details
- Make sure there is no regression
- Follow various Maker borrow and multiply positions, follow various AAVEV2 and V3 positions make sure all data is displayed correctly
- Screen with various multi protocol positions displayed in Follow Table 
![image](https://user-images.githubusercontent.com/6871923/224297903-66dcee26-029c-44e9-a61f-d40d80f2af95.png)


- Video recording, PS.: I'll check if error that occurred on position 99 is related just to this branch and address it


https://user-images.githubusercontent.com/6871923/224286874-f3e34c76-1392-4a22-aa7c-22b83e460ff2.mp4


- To test just API alone here's example correct request how to follow aave vault

```sh

curl --location --request POST 'http://localhost:3000/api/follow' \
--header 'Accept: */*' \
--header 'Accept-Language: en-US,en;q=0.9' \
--header 'Connection: keep-alive' \
--header 'Content-Type: application/json' \
--header 'Cookie: __adroll_fpc=dd5499e9cb811642e4079ce1e0c0d04e-1641986228346; __ar_v4=4PSXWPAQTJEIXJSPHJCRW4%3A20221016%3A99%7C5VWGFTJXUZF6BPUDCH3WWM%3A20221016%3A99; mp_b10b850880cb0a8557d878c2e6024d03_mixpanel=%7B%22distinct_id%22%3A%20%220xf39fd6e51aad88f6f4ce6ab8827279cfffb92266%22%2C%22%24device_id%22%3A%20%2217d95a5c4312bc-0c509204e1aa57-14241209-1fa400-17d95a5c432e3c%22%2C%22%24user_id%22%3A%20%220xf39fd6e51aad88f6f4ce6ab8827279cfffb92266%22%2C%22%24initial_referrer%22%3A%20%22%24direct%22%2C%22%24initial_referring_domain%22%3A%20%22%24direct%22%7D' \
--header 'Origin: http://localhost:3000' \
--header 'Referer: http://localhost:3000/aave/v3/78' \
--header 'Sec-Fetch-Dest: empty' \
--header 'Sec-Fetch-Mode: cors' \
--header 'Sec-Fetch-Site: same-origin' \
--header 'Sec-GPC: 1' \
--header 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36' \
--header 'authorization: Bearer $YOUR_TOKEN_HERE' \
--header 'sec-ch-ua: "Not_A Brand";v="99", "Brave";v="109", "Chromium";v="109"' \
--header 'sec-ch-ua-mobile: ?0' \
--header 'sec-ch-ua-platform: "Linux"' \
--data-raw '{"vault_id":44,"vault_chain_id":1,"protocol":"aavev3","proxy":"0x718c682efbC8edBA0d751Ae122dd7C6787b557eb","strategy":"wstETHeth"}'
```
